docs/getting_started/contribution_guide.rst

```diff
@@ -59,4 +59,126 @@ Kiso-testing and make sure that you understand your rights and obligations.
 Any contributions to Kiso-testing project code base needs to be licensed under
 EPL v2.0.
 
+What should I do before committing ?
+------------------------------------
+
+PEP8-Compliancy
+---------------
+In order to maintain clear and user-friendly project make sure, that your
+changes respect PEP8 standards. PEP8 is a guide that provides Python coding
+conventions (naming, indentation,...).
+Official document : https://peps.python.org/pep-0008/
+
+To make sure your changes are PEP8 Compliant, different tools exist to help you
+here:
+    - linter (applicable on IDE) :
+        show some warning directly on IDE
+    - pre-commit hook :
+        hook scripts that look after different issues (white space, trailing,...)
+
+
+Typography
+----------
+Most of the comments made during PR-Reviewing are about typography/misspelling
+mistakes. An easy way to avoid these is by running
+`codespell <https://pypi.org/project/codespell/>`_ on your written code.
+
+
+Function type hinting
+---------------------
+In kiso-testing, every implemented function must have annotations for its
+parameters and return types (type hints). This results in increased readability and
+therefore in easier comprehension of the code for any reader.
+
+    .. code:: python
+
+        def some_fun(some_dict_param: dict, some_string_param: str) -> list:
+
+    .. note:: As not every types are available in the builtins, or as it is
+        important to precise inner type you might import some from collections
+        module or typing
+
+    .. code:: python
+
+        from typing import List
+        from collections import namedtuple
+
+        def some_fun(
+            some_int_list_param: List[int], some_imported_type_param: namedtuple
+        ) -> list:
+
+
```

Comment:
To prevent circular dependencies 

```suggestion

.. code:: python

	from __future__ import annotations

	from typing import TYPE_CHECKING

	if TYPE_CHECKING:
		from Fruits import banana


	class BananaJuiceMaker:
		def __init__(self, banana: Banana) -> None:
			self.banana = banana

    ...
```